core/roboconf-target-embedded/src/main/java/net/roboconf/target/embedded/internal/EmbeddedHandler.java

```diff
@@ -51,26 +62,46 @@ public String getTargetId() {
 
 	@Override
 	public String createMachine( TargetHandlerParameters parameters ) throws TargetException {
-
+		if(ipTable == null && parameters != null && parameters.getTargetProperties() != null) {
+			String ips = parameters.getTargetProperties().get(IP_ADDRESSES);
+			if(ips != null) {
+				String iplist[] = ips.trim().split("\\s*,\\s*");
+				for(String ip : iplist) {
+					if(ipTable == null) ipTable = new HashMap<>();
+					ipTable.put(ip, false);
+				}
+			}
+		}
 		String machineId = parameters.getScopedInstancePath() + " (" + TARGET_ID + ")";
-		this.machineIdToRunning.put( machineId, Boolean.TRUE );
+		this.machineIdToRunning.put( machineId, "" );
 		return machineId;
 	}
 
 
 	@Override
 	public void terminateMachine( TargetHandlerParameters parameters, String machineId )
 	throws TargetException {
-		this.machineIdToRunning.remove( machineId );
+		String ip = this.machineIdToRunning.remove( machineId );
+		if(! Utils.isEmptyOrWhitespaces(ip)) releaseIpAddress(ip);
 	}
 
 
 	@Override
 	public void configureMachine( TargetHandlerParameters parameters, String machineId, Instance scopedInstance )
 	throws TargetException {
 
-		// It may require to be configured from the DM => add the right marker
-		scopedInstance.data.put( Instance.READY_FOR_CFG_MARKER, "true" );
+		String ip = acquireIpAddress();
+		if(! Utils.isEmptyOrWhitespaces(ip)) {
+			try {
+				sendConfiguration(ip, parameters);
+			} catch (IOException e) {
+				throw new TargetException(e);
+			}
+			this.machineIdToRunning.put(machineId, ip);
+		} else {
+			// It may require to be configured from the DM => add the right marker
+			scopedInstance.data.put( Instance.READY_FOR_CFG_MARKER, "true" );
```

Comment:
Shouldn't it be in a finally block?
Post configuration may be applied even after we updated the agent's configuration.